aloha/real_env.py

```diff
@@ -136,33 +127,56 @@ def setup_robots(self):
-        return np.concatenate(
-            [left_arm_qpos, left_gripper_qpos, right_arm_qpos, right_gripper_qpos]
-        )
+        # Initialize a list to hold the arm and gripper positions
+        qpos_list = []
+
+        # Iterate through all follower robots in the self.robots dictionary
+        for name, bot in self.robots.items():
+            if "follower" in name:
+                # Get the arm joint positions (first 6 joints)
+                arm_qpos = bot.core.joint_states.position[:6]
+                qpos_list.append(arm_qpos)
+
+                # Get the gripper joint position (7th joint) and normalize it
+                gripper_qpos = [FOLLOWER_GRIPPER_POSITION_NORMALIZE_FN(
+                    bot.core.joint_states.position[7])]
+                qpos_list.append(gripper_qpos)
+
+        # Concatenate all the positions into a single array
+        return np.concatenate(qpos_list)
 
     def get_qvel(self):
-        left_qvel_raw = self.recorder_left.qvel
-        right_qvel_raw = self.recorder_right.qvel
-        left_arm_qvel = left_qvel_raw[:6]
-        right_arm_qvel = right_qvel_raw[:6]
-        left_gripper_qvel = [FOLLOWER_GRIPPER_VELOCITY_NORMALIZE_FN(left_qvel_raw[7])]
-        right_gripper_qvel = [FOLLOWER_GRIPPER_VELOCITY_NORMALIZE_FN(right_qvel_raw[7])]
-        return np.concatenate(
-            [left_arm_qvel, left_gripper_qvel, right_arm_qvel, right_gripper_qvel]
-        )
+        # Initialize a list to hold the arm and gripper velocities
+        qvel_list = []
+
+        # Iterate through all follower robots in the self.robots dictionary
+        for name, bot in self.robots.items():
+            if "follower" in name:
+                # Get the arm joint velocities (first 6 joints)
+                arm_qvel = bot.core.joint_states.velocity[:6]
+                qvel_list.append(arm_qvel)
+
+                # Get the gripper joint velocity (7th joint) and normalize it
+                gripper_qvel = [FOLLOWER_GRIPPER_VELOCITY_NORMALIZE_FN(
+                    bot.core.joint_states.velocity[6])]
+                qvel_list.append(gripper_qvel)
+
+        # Concatenate all the velocities into a single array
+        return np.concatenate(qvel_list)
 
     def get_effort(self):
-        left_effort_raw = self.recorder_left.effort
-        right_effort_raw = self.recorder_right.effort
-        left_robot_effort = left_effort_raw[:7]
-        right_robot_effort = right_effort_raw[:7]
-        return np.concatenate([left_robot_effort, right_robot_effort])
+        # Initialize a list to hold the efforts for all arms and grippers
+        effort_list = []
+
+        # Iterate through all follower robots in the self.robots dictionary
+        for name, bot in self.robots.items():
+            if "follower" in name:
+                # Get the first 7 effort values (6 for arm, 1 for gripper)
+                robot_effort = bot.core.joint_states.effort[:7]
```

Comment:
We also need method to get gripper effort. I'll concatenate the arm and gripper efforts here.

Reply:
Added in https://github.com/Interbotix/interbotix_ros_toolboxes/pull/80/commits/d06005312c1db8cbb956ca05ecfb75f159fc19d4